pkg/controllers/clusterresourceplacement/placement_status_test.go

```diff
@@ -3123,405 +3163,1800 @@ func TestSetPlacementStatus(t *testing.T) {
+								Reason:             condition.ScheduleSucceededReason,
+								ObservedGeneration: crpGeneration,
+								LastTransitionTime: metav1.NewTime(currentTime),
+							},
+							{
+								Status:             metav1.ConditionTrue,
+								Type:               string(fleetv1beta1.ResourceWorkSynchronizedConditionType),
+								Reason:             condition.WorkSynchronizedReason,
+								ObservedGeneration: crpGeneration,
+								LastTransitionTime: metav1.NewTime(currentTime),
+							},
+							{
+								Status:             metav1.ConditionTrue,
+								Type:               string(fleetv1beta1.ResourcesDiffReportedConditionType),
+								Reason:             condition.DiffReportedStatusTrueReason,
+								ObservedGeneration: crpGeneration,
+								LastTransitionTime: metav1.NewTime(currentTime),
+							},
+						},
+					},
+					{
+						ClusterName:                        "member-2",
+						ApplicableClusterResourceOverrides: []string{"o-1", "o-2"},
+						ApplicableResourceOverrides: []fleetv1beta1.NamespacedName{
+							{
+								Name:      "override-1",
+								Namespace: "override-ns",
+							},
+							{
+								Name: "override-2",
+							},
+						},
+						Conditions: []metav1.Condition{
+							{
+								Status:             metav1.ConditionTrue,
+								Type:               string(fleetv1beta1.ResourceOverriddenConditionType),
+								Reason:             condition.OverriddenSucceededReason,
+								ObservedGeneration: crpGeneration,
+								LastTransitionTime: metav1.NewTime(currentTime),
+							},
+							{
+								Status:             metav1.ConditionTrue,
+								Type:               string(fleetv1beta1.ResourceRolloutStartedConditionType),
+								Reason:             condition.RolloutStartedReason,
+								ObservedGeneration: crpGeneration,
+								LastTransitionTime: metav1.NewTime(currentTime),
+							},
+							{
+								Status:             metav1.ConditionTrue,
+								Type:               string(fleetv1beta1.ResourceScheduledConditionType),
+								Reason:             condition.ScheduleSucceededReason,
+								ObservedGeneration: crpGeneration,
+								LastTransitionTime: metav1.NewTime(currentTime),
+							},
+							{
+								Status:             metav1.ConditionTrue,
+								Type:               string(fleetv1beta1.ResourceWorkSynchronizedConditionType),
+								Reason:             condition.WorkSynchronizedReason,
+								ObservedGeneration: crpGeneration,
+								LastTransitionTime: metav1.NewTime(currentTime),
+							},
+							{
+								Status:             metav1.ConditionUnknown,
+								Type:               string(fleetv1beta1.ResourcesDiffReportedConditionType),
+								Reason:             condition.DiffReportedStatusUnknownReason,
+								ObservedGeneration: crpGeneration,
+								LastTransitionTime: metav1.NewTime(currentTime),
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "ReportDiff apply strategy, one cluster has failed to report diff",
```

Comment:
add a test when the condition (which is before DiffReported) fails or unknown

Reply:
👌

Reply:
missed before, could you please add two more cases:
1. before the apply strategy is reportDiff and crp has reportDiff related conditions, CRP is updated to client-side apply and applied condition is false.
2. before the apply strategy is client-side apply and crp has applied and available condition, CRP is updated to the reportDiff and reportDiff is comleted.

Reply:
If we were to simulate apply strategy changes it will have to be done via integration/E2E tests; for the unit tests the method always processes one apply strategy type. Though we could verify the removal of condition types depending on the apply strategy type; the two cases added are:

a) clear `DiffReported` condition when the apply strategy is CSA/SSA
b) clear `Applied` and `Available` condition when the apply strategy is ReportDiff

Note that the Applied = False, no Available condition case is already covered in prev. test cases.